src/main/java/com/denizenscript/clientizen/events/KeyPressReleaseScriptEvent.java

```diff
@@ -3,52 +3,64 @@
 import com.denizenscript.denizencore.events.ScriptEvent;
 import com.denizenscript.denizencore.objects.ObjectTag;
 import com.denizenscript.denizencore.objects.core.ElementTag;
+import it.unimi.dsi.fastutil.ints.Int2ObjectMap;
+import it.unimi.dsi.fastutil.ints.Int2ObjectOpenHashMap;
+import net.minecraft.client.util.InputUtil;
 
-import java.util.Arrays;
-import java.util.Map;
-import java.util.stream.Collectors;
+public class KeyPressReleaseScriptEvent extends ScriptEvent {
 
-public class KeyPressReleaseEvent extends ScriptEvent {
+	public static KeyPressReleaseScriptEvent instance;
 
-	public static KeyPressReleaseEvent instance;
-
-	public KeyboardKeys key;
+	public Keys key;
 	public boolean pressed;
+	public KeyType type;
 
 
-	public KeyPressReleaseEvent() {
-		registerCouldMatcher("keyboard <'key'> pressed|released");
+	public KeyPressReleaseScriptEvent() {
+		registerCouldMatcher("key pressed|released|toggled");
```

Comment:
Since this is now also handling mouse events, perhaps the first arg could be `key|mouse|input` with `input` accepting both key and mouse events